include/kwk/utility/container/stride.hpp

```diff
@@ -167,24 +167,29 @@ namespace kwk
 
     /// Equality comparison operator
     template<auto Strides2>
-    constexpr bool operator==( stride<Strides2> const& other) const noexcept
+    constexpr bool operator==(stride<Strides2> const& other) const noexcept
     {
       return kumi::to_tuple(*this) == kumi::to_tuple(other);
     }
 
     /// Indexing interface
     template<std::convertible_to<size_type>... Is>
-    constexpr size_type linearize(Is... is) const noexcept
+    KWK_PURE KWK_FORCEINLINE constexpr
+    size_type linearize(Is... is) const noexcept
     requires( sizeof...(Is) <= static_order )
     {
-      return kumi::inner_product(*this, kumi::make_tuple(static_cast<size_type>(is)...), size_type{0});
+      return [=]<int...i>(std::integer_sequence< int, i... >)
```

Comment:
This is already the implementation in kumi:
https://github.com/jfalcou/kumi/blob/5ac06a388109b8a794f11197f3f0a5df9fc4ea58/src/kumi/algorithm/inner_product.hpp#L103

Reply:
yes but it needlesly adds the intermediate make_tuple step/overhead

Reply:
That's true.